Documentation/ExtensionArchitecture/Tutorials/Tea/Model.rst

```diff
@@ -0,0 +1,289 @@
+
+The TCA is a nested PHP array. In this example, we need the the following
+keys on the first level:
+
+:php:`ctrl`
+    Settings for the complete table, such as a record title, a label
+    for a single record, default sorting, and the names of some
+    internal fields.
+
+:php:`types`
+    We only have one type of tea record, however it is mandatory to
+    describe at least one type. Here we define the order in which
+    the fields are displayed in the backend.
+
+:php:`columns`
+    Here we define all fields that can be used for user input in the
+    backend.
+
+
+..  _extbase_tutorial_tea_model_ctrl:
+
+TCA :php:`ctrl` - Settings for the complete table
+-------------------------------------------------
+
+..  include:: /CodeSnippets/Tutorials/Tea/Configuration/TCA/TeaCtrl.rst.txt
+
+..  _extbase_tutorial_tea_model_ctrl_title:
+
+:php:`title`
+~~~~~~~~~~~~
+
+Defines the title used when we are talking about the table in the backend.
+It will be displayed on top of the list view of records in the backend
+and in backend forms.
+
+..  figure:: /Images/ManualScreenshots/ExtensionArchitecture/Tutorials/Tea/TeaTitle.png
+    :class: with-shadow
+
+    The **title** of the :sql:`tea` table.
+
+Strings starting with :php:`LLL:` will be replaced with localized text. See chapter
+:ref:`Extension localization <extension_localization>`. All other strings
+will be output as they are. This title will always be output as "Tea" without localization:
+
+.. code-block:: php
+   :caption: EXT:tea/Configuration/TCA/tx_tea_domain_model_product_tea.php
+
+   [
+       'ctrl' => [
+           'title' => 'Tea',
+       ],
+   ]
+
+..  _extbase_tutorial_tea_model_ctrl_label:
+
+:php:`label`
+~~~~~~~~~~~~
+
+The :php:`label` is used as name for a specific tea record. The name is used
+in listings and in backend forms:
+
+
+..  figure:: /Images/ManualScreenshots/ExtensionArchitecture/Tutorials/Tea/TeaLabel.png
+    :class: with-shadow
+
+    The **label** of a tea record.
+
+..  _extbase_tutorial_tea_model_ctrl_others:
+
+:php:`tstamp`, :php:`deleted`, ...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These fields are used to keep timestamp and status information for each record. You can read more about them in the `TCA Reference, chapter ctrl <t3tca:ctrl>`.
+
+..  _extbase_tutorial_tea_model_columns:
+
+TCA :php:`columns` - Defining the fields
+-------------------------------------------------
+
+All fields that can be changed in the TYPO3 backend or used in the Extbase
+model have to be listed here. Otherwise they will not be recognized by TYPO3.
+
+The :sql:`title` field is defined like this:
+
+..  include:: /CodeSnippets/Tutorials/Tea/Configuration/TCA/TeaColumnTitle.rst.txt
+
+The title of the field is displayed above the input field. The type is a (string)
+input field. The other configuration values influence display (size of the input
+field) and or processing on saving (:php:`'eval' => 'trim'` removes whitespace).
```

Comment:
I do not understand why, but this `:php:...` is not rendered as code.

Reply:
I think we need a space before the double columns